### PR TITLE
Add e2e tests for PAO mustgather

### PR DESCRIPTION
**Pull request by mrniranjan**
Signed-off-by: Niranjan M.R <mrniranjan@redhat.com>

### CODE REVIEW

**Review by ffromani — COMMENTED** [review]
looks like a nice start. We should probably squash all these commits together.
Do we want to run `must-gather` in the testsuite or we want somehow depend on the must-gather output already be available? (I'd prefer the former, at least optionally)

**Comment by mrniranjan** [comment]
@fromanirh i want to run it as part of downstream test suite.

**Comment by fromanirh** [comment]
> @fromanirh i want to run it as part of downstream test suite.

ok, but in general this won't prevent us to possibly call must-gather in the testsuite, will it?

**Comment by mrniranjan** [comment]
> > @fromanirh i want to run it as part of downstream test suite.
> 
> ok, but in general this won't prevent us to possibly call must-gather in the testsuite, will it?

No, Should i extend this test suite to call  mustgather from a specified image and then run the tests ?

File: test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+			workerRTNodes, err = nodes.GetByLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+			Expect(err).ToNot(HaveOccurred())
+			for _, node := range workerRTNodes {
+				nodeSpecificFiles := []string{
+					fmt.Sprintf("nodes/%s/%s_logs_kubelet.gz", node.Name, node.Name),
+					fmt.Sprintf("nodes/%s/lscpu", node.Name),
+					fmt.Sprintf("nodes/%s/lspci", node.Name),
+					fmt.Sprintf("nodes/%s/proc_cmdline", node.Name),
```
Comment (marioferh) [inline test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go:88]: maybe add here "sys/devices/system/cpu/offline" to confirm last must gather commit is working. Then I am creating a e2e to deploy performance profile with offlined cpus. Maybe you can use that one to check is the offline file has the right content.

**Comment by fromanirh** [comment]
> > > @fromanirh i want to run it as part of downstream test suite.
> > 
> > 
> > ok, but in general this won't prevent us to possibly call must-gather in the testsuite, will it?
> 
> No, Should i extend this test suite to call mustgather from a specified image and then run the tests ?

I think we should, but not in this PR. We can start simple.

**Comment by mrniranjan** [comment]
@fromanirh @marioferh I have squashed all the commits to one commit. Regarding offline cpus checking in mustgather. I will send a separate PR. Since the files  are in sysinfo.tgz. I need to figure out how to untar it in a test case.

**Comment by fromanirh** [comment]
> @fromanirh @marioferh I have squashed all the commits to one commit. Regarding offline cpus checking in mustgather. I will send a separate PR. Since the files are in sysinfo.tgz. I need to figure out how to untar it in a test case.

We do something very similar in ghw, this can be a good example: https://github.com/jaypipes/ghw/blob/main/pkg/snapshot/unpack.go#L65

File: test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	}
+
+	testutils.BeforeAll(func() {
+		if mustGatherFailed {
+			Skip("No mustgather directory provided, check MUSTGATHER_DIR environment variable")
+		} else {
+			mustgatherPathContent, err := ioutil.ReadDir(mustGatherPath)
+			Expect(err).To(BeNil(), "failed to read the Mustgather Directory %s: %v", mustGatherPath, err)
+			for _, file := range mustgatherPathContent {
+				if strings.Contains(file.Name(), "registry") {
```
Comment (marioferh) [inline test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go:45]: why registry? the idea is take sysinfo.tgz, extract it and validate files.

**Pull request closed without merging**